### ISSUE TITLE
Linux terminal like selection behavior

### ISSUE THREAD

**Issue by JayDoubleu** [microsoft/terminal#5943]
# Description of the new feature/enhancement

In Linux terminals the selection behaviour is what I consider "normal"
For example:
- The cursor within terminal is different
- Selecting empty lines selects whole lines
- The Fluent terminal guys almost* got it right: https://github.com/felixse/FluentTerminal

This is number one reason I dislike using powershell or windows terminal.
Now I have to use X11 to lunch terminator from WSL to use it in linux fashion. 

# Proposed technical implementation details (optional)
Make Linux user friendly "linux gui mode" or something similar.
This would include:
- Mouse selection behaviours ported to feel like in Linux terminals
- Right click context menu. ( Is copy and paste context menu too much to ask ) ?

**Comment by DHowett** [comment]
Thanks for filing!

This is a bucket issue, so I'm going to split it up and direct you to the relevant threads.

* Cursor: #5028 (PR) for issue #1441
* Context menu: #3337
* Selecting empty lines selects whole lines: oh man, this one makes me sad. Yes, it should 100% work like that. Because we're sharing some Windows Console infrastructure, we inherited its relative inability to tell how long a line is (it can do it, it's just annoyed about it.) I filed #5976 to follow up.

/dup #1441 #3337 #5976
